src/common/util.js

```diff
@@ -628,6 +628,20 @@ function isNonEmptyArray(object) {
   return Array.isArray(object) && object.length > 0;
 }
 
+/**
+ * @param {string} description
+ * @returns {(node: any) => symbol}
+ */
+function getGroupIdWithDescription(description) {
```

Comment:
better name?

Reply:
I don't like it been curried, how about just `getGroupId(node, 'typeParameters')`?

I understand that it will use a global cache, instead of cache for each type, but it seems safe for now.

Reply:
Ah, sorry, it seems you are getting different groupId on same node?

Reply:
Yes.

Reply:
Okay. I'm OKay with it.

Reply:
`createGroupIdMapper`